### ISSUE TITLE
[BUG] aria-invalid violation reported for input type="radio"

### ISSUE THREAD

**Issue by dszewczyk1** [IBMa/equal-access#580]
## What tool(s) are you using?
IBM Equal Access Accessibility Checker version 3.1.12 (Chrome)

- [ ] `accessibility-checker-engine`
- [ ] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description

We’re using external library which sets aria-invalid for input type=”radio”. Accessibility Checker reports violation like:
![image](https://user-images.githubusercontent.com/9110884/146355051-bf84853d-2a4a-4abe-b5bc-cff39d8ca2cf.png)

On W3C site you can see that aria-invalid is allowed for the "radio" type.
https://www.w3.org/TR/wai-aria-1.1/#radio

**Comment by tombrunet** [comment]
`aria-invalid` is deprecated in ARIA 1.2: https://www.w3.org/TR/wai-aria-1.2/#radio . It should no longer be used.

**Comment by tombrunet** [comment]
BTW, the reason it is deprecated for radio is that it was a global property, but was not intended to be used on all roles (radio included).
> The aria-invalid attribute is used to indicate that the value entered into an input field does not conform to the format expected by the application.

For instance, if you have a text field that's expecting an email and you don't provide text in an email format. Radio buttons are not freeform inputs, so it's not possible for a user to provide invalid input for a radio button.

Usually, if a radio button isn't a valid choice, that radio button should be disabled to prevent the user from choosing it.

**Comment by dszewczyk1** [comment]
Thank you very much for your prompt answer and in-depth explanation!

**Comment by tombrunet** [comment]
No problem - btw, if you have an example from another package that seems like it makes sense, let us know. We'll take a look. Sometimes scenarios crop up that we need to point the ARIA working group at to think through a bit more.
